templates/components/multiagent/typescript/workflow/agents.ts

```diff
@@ -22,30 +23,101 @@ const getQueryEngineTool = async () => {
   });
 };
 
+const getAvailableTools = async () => {
+  const configFile = path.join("config", "tools.json");
+  let toolConfig: any;
+  const tools: BaseToolWithCall[] = [];
+  try {
+    toolConfig = JSON.parse(await fs.readFile(configFile, "utf8"));
+  } catch (e) {
+    console.info(`Could not read ${configFile} file. Using no tools.`);
+  }
+  if (toolConfig) {
+    tools.push(...(await createTools(toolConfig)));
+  }
+
+  return tools;
+};
+
 export const createResearcher = async (chatHistory: ChatMessage[]) => {
+  const tools: BaseToolWithCall[] = [];
+  const queryEngineTool = await getQueryEngineTool();
+  if (queryEngineTool) {
+    tools.push(queryEngineTool);
+  }
+  const availableTools = await getAvailableTools();
+  // Add wikipedia, duckduckgo if they are available
+  for (const tool of availableTools) {
+    if (
+      tool.metadata.name === "wikipedia.WikipediaToolSpec" ||
```

Comment:
conceptually it makes more sense to add query engine tool to the set of available tools
```suggestion
  const availableTools = await getAvailableTools();
  // Add wikipedia, duckduckgo, and query engine if they are available
  for (const tool of availableTools) {
    if (
     tool.metadata.name === "queryenginetool" ||
      tool.metadata.name === "wikipedia.WikipediaToolSpec" ||
```